discovery/discovery.go

```diff
@@ -202,14 +205,13 @@ type FailedAttempt struct {
 	Error  error
 }
 
-func walker(out *Endpoints, attempts *[]FailedAttempt, testFn DiscoverWalkFunc) DiscoverWalkFunc {
-	return func(pre string, eps *Endpoints, err error) error {
+func walker(attempts *[]FailedAttempt, testFn DiscoverWalkFunc) DiscoverWalkFunc {
+	return func(pre string, dd *discoveryData, err error) error {
 		if err != nil {
 			*attempts = append(*attempts, FailedAttempt{pre, err})
 			return nil
 		}
-		out.Append(*eps)
```

Comment:
Yeah that's what I was thinking (and also that I was also missing something ;-). Sounds fair to me.